tests/phpunit/tests/user.php

```diff
@@ -915,6 +915,78 @@ public function test_wp_insert_user_should_reject_user_login_that_matches_existi
 		$this->assertSame( 'existing_user_email_as_login', $user_id->get_error_code() );
 	}
 
+	/**
+	 * Tests that `wp_update_user()` allows updating a user when
+	 * the `user_login` and `user_email` are the same.
+	 *
+	 * @ticket 57967
+	 *
+	 * @covers ::wp_update_user
+	 */
+	public function test_wp_update_user_should_allow_user_login_with_same_user_email() {
```

Comment:
_Before_ applying the fix in this PR, this test fails as an instance of `WP_Errror` is returned with the error message of `Sorry, that username is not available.` This is expected ✅